### DISCUSSION TITLE
Is there a way to disable update and create functions in a mongoose scope?

### DISCUSSION THREAD

**Discussion by vexCoder** [Automattic/mongoose#13429] [Q&A] [answered]
I have a script runner that uses mongoose, and i wanted to disable updating or creating. so the script wont accidentally overwrite data. is there a way to turn off updates/create in mongoose level?

**Comment by vkarpov15** [comment, marked as answer]
[Plugins](https://mongoosejs.com/docs/plugins.html) and [middleware](https://mongoosejs.com/docs/middleware.html). Below is a general idea.

```javascript
mongoose.plugin(schema => {
  schema.pre('save', () => { throw new Error('save not allowed'); });
  schema.pre(['updateOne', 'updateMany', 'findOneAndUpdate', 'findOneAndDelete', 'findOneAndReplace', 'deleteOne', 'deleteMany', 'replaceOne'], () => {
    throw new Error('Writes not allowed');
  });
});
```

Another approach would be to define a separate user on your database that only has readonly privileges, and use that user for scripting.
